src/components/bottom-sheet/accounts-bottom-sheet/accounts-bottom-sheet.tsx

```diff
@@ -4,46 +4,170 @@ import { withTheme, IThemeProps } from '../../../core/theme/with-theme';
 import stylesProvider from './styles';
 import { smartConnect } from '../../../core/utils/smart-connect';
 import BottomSheet from 'reanimated-bottom-sheet';
-import { AccountsScreen } from '../../../screens/accounts/accounts';
+import { Text } from '../../../library';
 import { BottomSheetHeader } from '../header/header';
+import { IAccountState, TokenType } from '../../../redux/wallets/state';
+import { setSelectedAccount, getBalance } from '../../../redux/wallets/actions';
+import { IReduxState } from '../../../redux/state';
+import {
+    getSelectedWallet,
+    getSelectedAccount,
+    getAccounts
+} from '../../../redux/wallets/selectors';
+import { connect } from 'react-redux';
+import { formatAddress } from '../../../core/utils/format-address';
+import { ListAccount } from '../../../screens/token/components/list-account/list-account';
+import { Blockchain } from '../../../core/blockchain/types';
+import { Amount } from '../../amount/amount';
+import { getBlockchain } from '../../../core/blockchain/blockchain-factory';
+import BigNumber from 'bignumber.js';
 
 interface IExternalProps {
     snapPoints: { initialSnap: number; bottomSheetHeight: number };
     onOpenStart: () => void;
     onCloseEnd: () => void;
 }
+export interface IReduxProps {
+    setSelectedAccount: typeof setSelectedAccount;
+    selectedAccount: IAccountState;
+    getBalance: typeof getBalance;
+    exchangeRates: any;
+    accounts: IAccountState[];
+}
+const mapStateToProps = (state: IReduxState) => {
+    return {
+        wallet: getSelectedWallet(state),
+        selectedAccount: getSelectedAccount(state),
+        exchangeRates: (state as any).market.exchangeRates,
+        accounts: getAccounts(state, getSelectedAccount(state).blockchain)
+    };
+};
+const mapDispatchToProps = {
+    setSelectedAccount,
+    getBalance
+};
 
 export class AccountsBottomSheetComponent extends React.Component<
-    IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>
+    IExternalProps & IReduxProps & IThemeProps<ReturnType<typeof stylesProvider>>
 > {
     public bottomSheet: any;
 
-    constructor(props: IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>) {
+    constructor(
+        props: IExternalProps & IReduxProps & IThemeProps<ReturnType<typeof stylesProvider>>
+    ) {
         super(props);
         this.bottomSheet = React.createRef();
     }
 
     public componentDidMount() {
         this.bottomSheet.current.props.onOpenStart();
         Platform.OS !== 'web' ? this.bottomSheet.current.snapTo(1) : null;
+        this.props.accounts.map(acc => {
+            this.props.getBalance(acc.blockchain, acc.address, undefined, true);
+        });
     }
 
+    public calculateBalance = (account: IAccountState) => {
```

Comment:
this should be transformed in an util since i think it's used in more than one places